packages/queries/src/QueryBuilderError.ts

```diff
@@ -0,0 +1,81 @@
+import { Query } from './types/types';
+import { RefOp } from './types/RefOp';
+
+export class QueryBuilderError extends Error {
+    constructor(
+        // TODO: extend the class to include the query. Blocker from doing
+        // this now, is that `AnyQuery`, still lives in the backend package
+        // public query: AnyQuery,
+        public type: string,
+        public message: string,
+    ) {
+        super(message);
+        Error.captureStackTrace(this, QueryBuilderError);
+    }
+
+    static createNestedQueryMissingRefOpWhereClauseError<DB>({
```

Comment:
wait why do we have this and `createNestedQueryInvalidRefOpWhereClauseError`?

Reply:
Invalid was one I created when I was working with the idea that we could check for incorrect RefOps... might still be useful in the future, so I kept it

Reply:
Don't, we can always re-build it. Unused code has a tendency to just take up space and never get used.